doc/source/contribution_and_api/unit_testing.rst

```diff
@@ -0,0 +1,171 @@
+is one of the checks that must be approved to merge the code.
+
+
+.. figure:: ../images/codecov_increase.png
+    :width: 400pt
+
+
+Coverage example
+----------------
+
+Let's show with an example how the coverage works.
+Let's suppose that we have the following library:
+
+**My awesome library**
+
+
+.. code:: python
+
+    def get_report_colors(theme):
+
+        if theme == 'weather':
+            colors = ["blue", "lightblue", "grey"]
+        elif theme == 'traffic':
+            colors = ["red", "orange", "yellow"]
+        else:
+            colors = ["red", "blue", "green"]
+        
+        return colors
+
+
+**Tests**
+
+You can opt to run the tests with the following configuration:
+
+.. code:: python
+
+   def test_get_report_colors():
+
+        assert get_report_colors('weather') == ["blue", "lightblue", "grey"]
+        assert get_report_colors('traffic') == ["red", "orange", "yellow"]
+        assert get_report_colors('other') == ["red", "blue", "green"]
+
+
+Or, if the function is a bit more complex, you split the case in different tests:
+
+.. code:: python
+
+    def test_get_report_colors_weather():
+
+        assert get_report_colors('weather') == ["blue", "lightblue", "grey"]
+
+    def test_get_report_colors_traffic():
+
+        assert get_report_colors('traffic') == ["red", "orange", "yellow"]
+
+    def test_get_report_colors_other():
+
+        assert get_report_colors('other') == ["red", "blue", "green"]
+
+
+In any of the cases, the coverage will be 100% for the function, but the second case will
+be more useful to debug the function.
+
+
+Test Validation GitHub workflow
+-------------------------------
+
+Unit tests and integration tests are part of the Continuous Integration (CI) approach. 
+Combined with the Continuous Development (CD) approach, they form the CI/CD approach. 
+This approach provides continuous integration (CI) of the newly added code by
+automating its testing, monitoring and deployment which allow us a continuous
+deployment (CD) throughout the application lifecycle.
+
+.. figure:: ../images/cicd.jpg
+    :width: 300pt
+
+Create a unit test 
+------------------
+
+In the PyMAPDL repository, tests run using `pytest <pytest_>`_. 
+
+To create a pytest file, its name must be in the form ``test_XXX.py``  where ``XXX``
+can be the function/method/class you are testing or some other explicative
+name. Please keep in mind that ``pytest`` can filter the tests to run in the command
+line using the argument ``-k``. For more information visit `pytest usage <pytest_usage_>`_.
+
+Here is some advice to follow when creating good unit tests: 
+
+1. The test method names are long and descriptive.
+2. The tests cover all the code implementation. You can check it with **Codecov**.
+3. The tests should return the same result each time. 
+4. The tests are independent.
+5. Each test verifies only one part of the code at a time.
+6. The tests should be as short and fast as possible.
```

Comment:
```suggestion
Here are some guidelines for creating good unit tests: 

- Assign long and descriptive names to tests.
- Use the `Codecov <codecov_>`_ tool to ensure all implemented code is tested.
- Check that tests return the same results each time. 
- Verify that tests are independent.
- Write tests that verify only one part of the code at a time.
- Make tests as short and fast as possible.
```
I changed these to a bulleted list as the entries don't have to be performed in sequential order.